### ISSUE TITLE
curly rule behavior with "multi-nest" option seems strange with 'do'

### ISSUE THREAD

**Issue by aparajita** [eslint/eslint#5487]
**What version of ESLint are you using?**

2.3.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**

Espree

**What did you do? Please include the actual source code causing the issue.**

```js
/* eslint curly: [2, multi-nest] */
do {
    doSomething();
} while (test());
```

**What did you expect to happen?**

I did not expect the curly rule to complain.

**What actually happened? Please include the actual, raw output from ESLint.**

`error    Unnecessary { after 'do'`

I understand that this is the documented behavior, but it seems very strange to me to enforce this rule on `do/while`, because it means doing something like this:

```js
do
   doSomething();
while (test());
```

This somehow seems very strange and error-prone. It's very different than the other control statements which require only one line. I'm wondering if `do/while` should be excluded or if an "allow" option could exclude it.

**Comment by michaelficarra** [comment]
Just put it on one line.

**Comment by aparajita** [comment]
That breaks the stylistic rules.

I'll live with the rule off or get used to using do/while without braces.

**Comment by platinumazure** [comment]
What about the always option? Is there a time when you don't want curlies?
